src/applications/simple-forms/21-0966/config/submit-transformer.js

```diff
@@ -1,12 +1,25 @@
+import { cloneDeep } from 'lodash';
 import sharedTransformForSubmit from '../../shared/config/submit-transformer';
 import {
   veteranBenefits,
   survivingDependentBenefits,
 } from '../definitions/constants';
+import { preparerIsVeteranAndHasPrefill } from './helpers';
 
 export default function transformForSubmit(formConfig, form) {
+  const formData = cloneDeep(form.data);
+
+  if (preparerIsVeteranAndHasPrefill({ formData })) {
+    formData.veteranFullName = formData['view:veteranPrefillStore'].fullName;
+  }
+
+  // remove all view: fields
+  Object.keys(formData)
+    .filter(key => key.startsWith('view:'))
+    .forEach(key => delete formData[key]);
```

Comment:
This will unapologetically remove all key/value pairs that start with `view:` in any spot in `formData`